attribute-service-api/src/main/proto/org/hypertrace/core/attribute/service/v1/attribute_metadata.proto

```diff
@@ -98,6 +98,7 @@ message AttributeMetadata {
   map<string, AttributeSourceMetadata> metadata = 13;
   string id = 14;
   bool groupable = 15;
+  Projection definition = 16;
```

Comment:
@aaron-steinfeld one fundamental question is, we defines this projection? Should it all come via the config bootstrapping? That might mean putting lot of span key paths in these configs.

Reply:
That was the idea that I was thinking - unifying the definition of an attribute in one spot, spanning from the user's point of view all the way down to the input data. Without this, using attributes becomes different for raw attributes vs derived/mutable attributes and raw or enriched attributes additions/changes require code level changes. At least that's based on my current understanding.